package/MDAnalysis/lib/formats/xtc.pyx

```diff
@@ -0,0 +1,305 @@
+                             'supplied {}'.format(mode))
+        self.mode = mode
+
+        if self.mode == 'r':
+            if not exists(self.fname):
+                raise IOError('File does not exists: {}'.format(self.fname))
+
+            return_code = xdrlib.read_xtc_natoms(fname, &self.n_atoms)
+
+            if return_code != 0:
+
+                raise IOError('XTC read error: {}'.format(return_code))
+            if self.n_atoms <= 0:
+                raise IOError('Couldn\'t read number of atoms: {}'.format(
+                    self.fname))
+
+        self.xfp = xdrlib.xdrfile_open(fname, opening_mode)
+        if self.xfp is NULL:
+            raise IOError('error opening xtf file: {}'.format(self.fname))
+        self.is_open = True
+
+    def __enter__(self):
+        """Support context manager"""
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Support context manager"""
+        self.close()
+        # always propagate exceptions forward
+        return False
+
+    def __iter__(self):
+        self.close()
+        self.open(self.fname, self.mode)
+        return self
+
+    def __next__(self):
+        if self.reached_eof:
+            raise StopIteration
+        return self.read()
+
+    # TODO: cache result in file
+    def __len__(self):
+        if not self.is_open:
+            raise RuntimeError('No file currently opened')
+
+        cdef int old_frame = self.current_frame
+        try:
+            while True:
+                self.read()
+        except StopIteration:
+            pass
+        cur_frame = self.current_frame
+
+        self.seek(old_frame)
+        return cur_frame
+
+    # TODO: optimize with direct jump in file
+    def seek(self, frame):
+        """Seek to Frame
+
+        Parameters
+        ----------
+        frame: int
+            wind the file to given frame
+
+        Raises
+        ------
+        RuntimeError
+            If you seek for more frames then are available
+        """
+        self.close()
+        self.open(self.fname, self.mode)
+        while self.current_frame < frame:
+            try:
+                self.read()
+            except StopIteration:
+                raise RuntimeError('Trying to seek over max number of frames')
+
+    def tell(self):
+        """Get current frame"""
+        return self.current_frame
+
+    def offsets(self):
+        if not self.is_open:
+            return np.array([])
+        cdef int n_frames = 0
+        cdef int est_nframes = 0
+        cdef xdrlib.int64_t* offsets = NULL
+
+        xdrlib.read_xtc_n_frames(self.fname, &n_frames, &est_nframes, offsets);
+
+        cdef np.ndarray dims = np.array([est_nframes], dtype=np.int64)
+        cdef np.ndarray _offsets = ptr_to_ndarray(<void*> offsets, dims, np.NPY_INT64)
+        print("read {} frames, estimated {}".format(n_frames, est_nframes));
+        print("first offset = {}".format(offsets[0]))
```

Comment:
I checked the memory is allocated in the xdrlib but the pointer at the cython level is never updated.